cmd/kubeadm/app/phases/upgrade/compute.go

```diff
@@ -112,13 +112,17 @@ func GetAvailableUpgrades(versionGetterImpl VersionGetter, experimentalUpgradesA
 		return nil, err
 	}
 
+	if !etcdStatus.VersionsMatch() {
+		return nil, fmt.Errorf("etcd cluster has nodes with mismatched versions: %v", etcdStatus.Versions)
+	}
+
 	// Construct a descriptor for the current state of the world
 	beforeState := ClusterState{
 		KubeVersion:     clusterVersionStr,
 		DNSVersion:      dns.GetDNSVersion(clusterVersion, ActiveDNSAddon(featureGates)),
 		KubeadmVersion:  kubeadmVersionStr,
 		KubeletVersions: kubeletVersions,
-		EtcdVersion:     etcdStatus.Version,
+		EtcdVersion:     etcdStatus.Versions[0],
```

Comment:
This would be more useful if it listed output like we do for kubelet. That shows the number of nodes at which version e.g.  `2 x 1.9.5, 1 x 1.9.6`